### PR TITLE
associate *.c++ & *.h++ files with c++

### LINKED ISSUE #104320: add file associations for ".c++" and ".h++" file extensions
i think vscode should map .c++ and .h++ file extensions to c++ by default. now, i know c++ is a language that has lots of file extensions associated with it, so it would make sense not doing it for all of them, but i'd argue .c++ and .h++ should be added because it would bring better compatibility with compilers, as some of them, like gcc, recognise them by default, and seti-ui, vscode's default file theme, also identifies them already. besides, i also propose making windows computers recognise those file extensions, and the only way for a user to do it is using the registry editor, which can be very dangerous if you don't really know what you're doing.

plus, there are thousands of files on github alone that have .c++ and .h++ as extensions ([source](https://github.com/search?q=%22include%22+extension%3Ac%2B%2B+extension%3Ah%2B%2B&type=Code)) (i know that number is rather small compared to other files extensions, especially .cpp and .hpp, but it's still an expressive number that i don't think should be ignored.)

### PR DESCRIPTION
**Pull request by therloux**
This PR fixes #104320 

i had to update manualy a .iss file (not so sure i should have -- maybe there's a script that does that automatically à la `update-icon-theme`?) so appropriate registry keys could be treated the same as `.cpp` & `.h`/`.hpp` files on windows. i also added `".hpp"` to the cpp extension so it can be recognised as a c++ file inside vscode. lastly i changed a regex in theme-seti's `update-icon-theme.js` script file because, even though seti-ui recognised `.h++`, the script ignored it. that's why i also had to update the icon themes.

i hope those were all of the changes i had to make. maybe you should also add more file extensions. i know there's a lot of them but you should at least add those that are recognised by most compilers by default, so there'd be better compatibility between vscode and c++ compilers (eg. gcc automatically recognises `.cp` and `.hp` as c++ files, but vscode doesn't).

### CODE REVIEW

**Comment by aeschli** [comment]
@amazingcaio Thanks a lot! Can you remove the seti-font update? I'd rather do that separately. The seti update-script change is fine.

The changes for the language is also fine.

@joaomoreno Do you approve the code.iss changes?

**Comment by amazingcaio** [comment]
@aeschli done. as you can see, i reverted the update but then i manually edited `vs-seti-icon-theme.json` to include `.h++` files because i think it's necessary for this pull request. i hope it's okay, but if you don't want me to touch that file at all, i can remove those changes.

**Comment by aeschli** [comment]
Better remove the `vs-seti-icon-theme.json` change as well. I'll soon regenerate the seti theme so you will get that change.

**Comment by amazingcaio** [comment]
@aeschli sure :) sorry for not doing it when you first asked.

**Comment by amazingcaio** [comment]
apparently i missed a file. now macos is supposed to recognise *.c++ and *.h++ just like *.cpp and *.hpp

i also noticed there's one other file that should change, `extensions/search-result/syntaxes/generateTMLanguage.js` (for syntax highlighting in search result tabs), but i'm confused, as all extensions aren't included in the file (e.g. "cxx", "hxx", and "hh" are missing), and .hpp is treated as an objective-c++ file. i suppose not including those file extensions was a mistake, and .hpp has objective-c++ syntax highlighting because it's also used in that language, whose syntax encompasses that of c++.

however, following that logic, shouldn't it apply to .h files as well? currently .h files have objective-c syntax highlighting. wouldn't it be better to treat them as objective-c++?

@JacksonKearl, you're the one who created those files. do you mind explaining it, so we could decide what to do? thx

**Comment by JacksonKearl** [comment]
@amazingcaio I'd say for this PR adding `cxx`, `hxx`, `hh`, `c++`, and `h++` mapping to ObjC++ makes sense. I don't think we should treat plain `.h` as C++ because C++ isn't exactly a superset of C, unlike the Objective-*'s. (Similarly, `.c` is not treated at C++)

Though I will admit that there wasn't terribly much thought put into programming language theory when constructing that mapping, it's really more of a best-effort approach. 🙂

**Comment by amazingcaio** [comment]
@JacksonKearl i see your point. so i suppose the missing extensions should be added based on the extensions already in the file then. should i change that file for this pull request? i could add all of the missing extensions for c++, or maybe add .c++ & .h++ only, as i believe this would be part of this pr (someone else would then have to update that file with the other extensions), or i could not change those files at all, and someone else (maybe you) could add all of those missing file extensions.

**Comment by JacksonKearl** [comment]
I'll just add them all and rebuild the grammar, this PR can stay as-is re. search editors.

**Comment by aeschli** [comment]
I already pushed the changes for the file association and the seti script for August. Thanks a lot @amazingcaio !
@joaomoreno just returned from vacation and can hopefully review the rest soon.

**Comment by joaomoreno** [comment]
Thanks!

**Comment by ret394** [comment]
I hope you did not incorporate the '.cp and .hp' file extensions as suggested by the author in the first post, because it'll be in version 1.53?

Saved you (vscode team), actual owners of the code.

you're welcome, as always. all the best. 👍
don't say '... passim' to folks on Github or other sites.

[ret394](https://github.com/ret394)

**Comment by ret394** [comment]
---

Congrats @amazingcaio for getting it right in the first PR on many accounts. A rare sight in general and on this repo.

Thanks a lot.

File: extensions/cpp/package.json [hunk 1]
```diff
@@ -34,8 +34,8 @@
 					".c++",
 					".hpp",
 					".hh",
-					".hxx",
-					".h++",
+          ".hxx",
+          ".h++",
 					".h",
```
Comment (ret394) [inline extensions/cpp/package.json:39]: To confirm, this is in 1.50, right. Well, one less thing to put in for C++ environs.

thanks

**Pull request merged**